Tasks/PowerShellOnTargetMachines/PowerShellOnTargetMachines.ps1

```diff
@@ -256,20 +274,33 @@ function Get-ResourcesProperties
     return $resourcesPropertyBag
 }
 
-$connection = Get-VssConnection -TaskContext $distributedTaskContext
+try
+{
+  $connection = Get-VssConnection -TaskContext $distributedTaskContext
 
-Write-Verbose "Starting Register-Environment cmdlet call for environment : $environmentName with filter $machineFilter" -Verbose
-$environment = Register-Environment -EnvironmentName $environmentName -EnvironmentSpecification $environmentName -UserName $adminUserName -Password $adminPassword -WinRmProtocol $protocol -TestCertificate ($testCertificate -eq "true") -Connection $connection -TaskContext $distributedTaskContext -ResourceFilter $machineFilter
-Write-Verbose "Completed Register-Environment cmdlet call for environment : $environmentName" -Verbose
+  Write-Verbose "Starting Register-Environment cmdlet call for environment : $environmentName with filter $machineFilter" -Verbose
+  $environment = Register-Environment -EnvironmentName $environmentName -EnvironmentSpecification $environmentName -UserName $adminUserName -Password $adminPassword -WinRmProtocol $protocol -TestCertificate ($testCertificate -eq "true") -Connection $connection -TaskContext $distributedTaskContext -ResourceFilter $machineFilter
+  Write-Verbose "Completed Register-Environment cmdlet call for environment : $environmentName" -Verbose
 
-Write-Verbose "Starting Get-EnvironmentResources cmdlet call on environment name: $environmentName" -Verbose
-$resources = Get-EnvironmentResources -EnvironmentName $environmentName -TaskContext $distributedTaskContext
-if ($resources.Count -eq 0)
-{
-    throw (Get-LocalizedString -Key "No machine exists under environment: '{0}' for deployment" -ArgumentList $environmentName)
+  Write-Verbose "Starting Get-EnvironmentResources cmdlet call on environment name: $environmentName" -Verbose
+  $resources = Get-EnvironmentResources -EnvironmentName $environmentName -TaskContext $distributedTaskContext
+  if ($resources.Count -eq 0)
+  {
+      Write-Telemetry "PREREQ002"
+      throw (Get-LocalizedString -Key "No machine exists under environment: '{0}' for deployment" -ArgumentList $environmentName)
+  }
+
+  $resourcesPropertyBag = Get-ResourcesProperties -resources $resources
 }
+catch
```

Comment:
We should not I believe. We will rethrow the exception as it is. Why would we need to add ErrorRecord.FullyQualifiedErrorId? I thought the original exception ill contain it.